platform/jvm/Options.kt

```diff
@@ -0,0 +1,224 @@
+import com.vaticle.bazel.distribution.common.util.PropertiesUtil.getStringOrNull
+import com.vaticle.bazel.distribution.common.util.PropertiesUtil.requireString
+import com.vaticle.bazel.distribution.platform.jvm.CommandLineParams.Keys.APPLE_CODE_SIGNING_CERT_PASSWORD
+import com.vaticle.bazel.distribution.platform.jvm.CommandLineParams.Keys.APPLE_ID
+import com.vaticle.bazel.distribution.platform.jvm.CommandLineParams.Keys.APPLE_ID_PASSWORD
+import com.vaticle.bazel.distribution.platform.jvm.JVMPlatformAssembler.logger
+import com.vaticle.bazel.distribution.platform.jvm.Logging.LogLevel
+import com.vaticle.bazel.distribution.platform.jvm.Logging.Logger
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.APPLE_CODE_SIGN
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.APPLE_CODE_SIGNING_CERT_PATH
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.APPLE_DEEP_SIGN_JARS_REGEX
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.COPYRIGHT
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.CREATE_SHORTCUT
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.ICON_PATH
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.DESCRIPTION
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.IMAGE_FILENAME
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.IMAGE_NAME
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.JARS_PATH
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.VENDOR
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.JDK_PATH
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.LICENSE_PATH
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.LINUX_APP_CATEGORY
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.LINUX_MENU_GROUP
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.MAC_APP_ID
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.MAC_ENTITLEMENTS_PATH
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.MAIN_CLASS
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.MAIN_JAR_PATH
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.OUTPUT_ARCHIVE_PATH
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.LOG_SENSITIVE_DATA
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.SOURCE_FILENAME
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.VERBOSE
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.VERSION_FILE_PATH
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.WINDOWS_MENU_GROUP
+import com.vaticle.bazel.distribution.platform.jvm.Options.Keys.WINDOWS_WIX_TOOLSET_PATH
+import java.io.File
+import java.io.FileInputStream
+import java.util.Properties
+
+data class Options(val logging: Logging, val input: Input, val image: Image, val launcher: Launcher, val output: Output) {
```

Comment:
The Options object is a simple structure that just holds a number of fixed strings and stuff, so it's quite a natural data class.

To give it some kind of structure, we create inner data classes that contain the various building blocks of the overall configuration object.